lib/infrastructure/repayments/change_repayment/change_repayment_middleware.dart

```diff
@@ -0,0 +1,29 @@
+import 'package:redux/redux.dart';
+import 'package:solarisdemo/infrastructure/repayments/change_repayment/change_repayment_service.dart';
+import 'package:solarisdemo/redux/app_state.dart';
+import 'package:solarisdemo/redux/repayments/change_repayment/change_repayment_action.dart';
+
+class ChangeRepaymentMiddleware extends MiddlewareClass<AppState> {
+  final ChangeRepaymentService _changeRepaymentService;
+
+  ChangeRepaymentMiddleware(this._changeRepaymentService);
+
+  @override
+  call(Store<AppState> store, action, NextDispatcher next) async {
+    next(action);
+
+    if (action is UpdateChangeRepaymentAction) {
+      store.dispatch(ChangeRepaymentLoadingAction());
+
+      final response = await _changeRepaymentService.updateChangeRepayment(
+        user: action.user.cognito,
+        fixedRate: action.fixedRate,
+      );
+      if (response is ChangeRepaymentSuccessResponse) {
+        store.dispatch(UpdateChangeRepaymentAction(user: action.user, fixedRate: action.fixedRate));
```

Comment:
In general aici vrei sa updatezi cu detalii din raspuns nu cu detalii din action. Daca raspunsul e doar cu status code si nu are body, atunci il mapezi tu in service ca sa il ai in middleware.
A doua chestie, nu ai nevoie sa trimiti userul mai departe pentru ca teoretic il ai in auth state oriunde ai nevoie sa il folosesti in screenuri. Il punem in actions doar pentru ca avem nevoie de el in request, dar mai departe nu ai de ce sa il propagi.